engine/classes/Elgg/Application.php

```diff
@@ -408,6 +408,13 @@ public function run() {
 			return true;
 		}
 
+		if (0 === strpos($path, '/refresh-token')) {
+			$this->loadCore();
+			$this->services->session->start();
```

Comment:
I think our implementation reads from the database.
On Apr 13, 2016 4:21 PM, "Steve Clay" <notifications@github.com> wrote:

> In engine/classes/Elgg/Application.php
> <https://github.com/Elgg/Elgg/pull/9675#discussion_r59556136>:
>
> > @@ -408,6 +408,13 @@ public function run() {
> >  			return true;
> >  		}
> >
> > +		if (0 === strpos($path, '/refresh-token')) {
> > +			$this->loadCore();
> > +			$this->services->session->start();
>
> It's fine for now, but I feel like this shouldn't be necessary. Like
> ElggSession get()/set() should make sure of that.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9675/files/cc9cccf1ef891e8b9fcf3e04c0523abf2b7b7c5d#r59556136>
>